initial sonar config

ottojo opened the pull request:
* Static code analysis via sonarcloud.io
* Coverage enabled for entire debug build
* Move more options to top-level cmake

ottojo committed:
initial sonar config
* Static code analysis via sonarcloud.io
* Coverage enabled for entire debug build
* Move more options to top-level cmake

ottojo commented:
Sonarclound brauch hier noch Berechtigungen für die Org, und das Secret `SONAR_TOKEN` (von sonarcloud) muss in diesem Repo eingetragen werden.

ottojo committed:
add sonar-project.properties

ottojo committed:
remove cpplint config

csacro commented:
> 
> 
> Sonarclound brauch hier noch Berechtigungen für die Org, und das Secret `SONAR_TOKEN` (von sonarcloud) muss in diesem Repo eingetragen werden.

Du solltest jetzt Zugriff aufs Repo haben

ottojo commented:
> > Sonarclound brauch hier noch Berechtigungen für die Org, und das Secret `SONAR_TOKEN` (von sonarcloud) muss in diesem Repo eingetragen werden.
> 
> Du solltest jetzt Zugriff aufs Repo haben

Sonar sollte Org-weit installiert werden, das kann nur der Owner @Marco5de

Marco5de commented:
> > > Sonarclound brauch hier noch Berechtigungen für die Org, und das Secret `SONAR_TOKEN` (von sonarcloud) muss in diesem Repo eingetragen werden.
> > 
> > 
> > Du solltest jetzt Zugriff aufs Repo haben
> 
> Sonar sollte Org-weit installiert werden, das kann nur der Owner @Marco5de

Solltest jetzt owner sein

ottojo committed:
fix org and repo name

ottojo committed:
fix project key, add multithreaded analysis

ottojo committed:
split CI into multiple workflows
this allows sonar to only run on PRs towards develop.
Sonar does not seem to support analysis of gitflow based projects:
https://community.sonarsource.com/t/having-multiple-long-lived-branches-on-github/6280/5
The decision is to only analyze the develop branch and PRs against it.

ottojo committed:
use gcno and gcov data to maybe fix coverage reports for shared library

ottojo committed:
remove duplicate sonar config from CI
(should have been removed in 045124cf572a7b9f936c3b118dadc3dc98b509d3 )

ottojo commented:
Coverage ist noch ein bisschen kaputt, ich versuch das noch zu fixen (Ich glaube nur auf der analysis von Develop, die ich manuell machen musste... vielleicht fixt sich das von selbst)

ottojo commented:
YAY, coverage ist gefixt. Problem war, dass auf develop hier: https://github.com/SoPra-Team-17/LibCommon/pull/20/files#diff-95e351a3805a1dafa85bf20b81d086e6L12 die `CMAKE_CX_FLAGS` zurückgesetzt wurden, und damit coverage deaktiviert wurde.

Marco5de reviewed (approved):
soweit ich das nachvollziehen kann, sieht das gut aus.
